envision/server.py

```diff
@@ -365,8 +365,13 @@ def initialize(self, path_map: Dict[str, Union[str, Path]] = {}):
 
     async def get(self, id_):
         """Serve a resource requested by id."""
-        if id_ not in self._path_map or not self._path_map[id_].exists():
-            raise tornado.web.HTTPError(404)
+        if id_ not in self._path_map:
+            raise tornado.web.HTTPError(404, f"Map resource {id_} not found")
+
+        if not self._path_map[id_].exists():
```

Comment:
The type hint for `_path_map` says its values can be `str` or `Path`. It seems that calling `exists()` assumes it is a `Path`.